git-webkit pr crashes when run from detached HEAD

kmiller68 opened the pull request:
#### 3cf11f08a646410a517ce42ca33c8ca7367ffda5
<pre>
    git-webkit pr crashes when run from detached HEAD
    <a href="https://bugs.webkit.org/show_bug.cgi?id=240468">https://bugs.webkit.org/show_bug.cgi?id=240468</a>

    Reviewed by Jonathan Bedard.

    repository.branch is None when on a detached HEAD. This causes us to throw an Error when performing a regex on the branch name.
    This patch treats a detached HEAD the same as being on main.

    * Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/pull_request.py:
    (PullRequest.pull_request_branch_point):

Canonical link: <a href="https://commits.webkit.org/250961@main">https://commits.webkit.org/250961@main</a>
git-svn-id: <a href="https://svn.webkit.org/repository/webkit/trunk@294805">https://svn.webkit.org/repository/webkit/trunk@294805</a> 268f45cc-cd09-0410-ab3c-d52691b4dbfc
</pre>

JonWBedard reviewed (approved):
Will need to be squashed, though.

zdobersek committed:
[Linux] GBMBufferSwapchain should only conditionally create linear-formatted buffer objects
https://bugs.webkit.org/show_bug.cgi?id=240637

Reviewed by Adrian Perez de Castro.

Spawning new gbm_bo objects should not default to requesting linear-storage
buffers. Instead, these should only be created when necessary, e.g. when
copying software-decoded video data into these buffers.

By default, no GBM flags are used. When required, the linear-storage requirement
is enscribed in the new flags field of the GBMBufferSwapchain::BufferDescription
object and acted upon in the GBMBufferSwapchain::getBuffer() call, using the
GBM_BO_USE_LINEAR flag in the gbm_bo_create() call.

The DMABuf-specific sink in MediaPlayerPrivateGStreamer is the only place at the
moment where linear-storage buffers are used since software-decoded material
originates here and is stored in linear memory.

* Source/WebCore/platform/graphics/gbm/GBMBufferSwapchain.cpp:
(WebCore::GBMBufferSwapchain::getBuffer):
* Source/WebCore/platform/graphics/gbm/GBMBufferSwapchain.h:
* Source/WebCore/platform/graphics/gbm/GraphicsContextGLGBM.cpp:
(WebCore::GraphicsContextGLANGLE::makeContextCurrent):
* Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp:
(WebCore::MediaPlayerPrivateGStreamer::pushDMABufToCompositor):

Canonical link: https://commits.webkit.org/250959@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@294803 268f45cc-cd09-0410-ab3c-d52691b4dbfc

alanbaradlay committed:
Expand flex-grow support
https://bugs.webkit.org/show_bug.cgi?id=240898

Reviewed by Antti Koivisto.

This patch add support for flex-grow. This property defines how flex items take over the available space.

* Source/WebCore/layout/formattingContexts/flex/FlexFormattingContext.cpp:
(WebCore::Layout::FlexFormattingContext::computeLogicalWidthForFlexItems):
(WebCore::Layout::FlexFormattingContext::layoutInFlowContentForIntegration):
* Source/WebCore/layout/formattingContexts/flex/FlexFormattingContext.h:

Canonical link: https://commits.webkit.org/250960@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@294804 268f45cc-cd09-0410-ab3c-d52691b4dbfc

kmiller68 committed:
git-webkit pr crashes when run from detached HEAD
    https://bugs.webkit.org/show_bug.cgi?id=240468

    Reviewed by Jonathan Bedard.

    repository.branch is None when on a detached HEAD. This causes us to throw an Error when performing a regex on the branch name.
    This patch treats a detached HEAD the same as being on main.

    * Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/pull_request.py:
    (PullRequest.pull_request_branch_point):

Canonical link: https://commits.webkit.org/250961@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@294805 268f45cc-cd09-0410-ab3c-d52691b4dbfc
